### PR TITLE
En_Bb (Blue Bubble) OK and documented. object_bb documented

### PR DESCRIPTION
**Pull request by tom-overton**
Before opening this PR, ensure the following:
- `./format.sh` was run to apply standard formatting.
- `make` successfully builds a matching ROM.
- No new compiler warnings were introduced during the build process.
    - Can be verified locally by running `tools/warnings_count/check_new_warnings.sh`
- New variables & functions should follow standard naming conventions.
- Comments and variables have correct spelling.
---

I'm not 100% sure if that `EnBbBodyPartDrawStatus` enum should exist, or whether I should just use raw int values instead. I also don't know if there is a better name for that; OoT's BodyBreak system just calls this `val`. Other than that, this is pretty straightforward.

### CODE REVIEW

**Review by AngheloAlf — APPROVED** [review]

File: src/overlays/actors/ovl_En_Bb/z_en_bb.c [hunk 1]
```diff
@@ -36,118 +47,667 @@ const ActorInit En_Bb_InitVars = {
...
+ * Checks to see if the Bubble is touching a wall. If it is, and if the
+ * Bubble is facing directly "into" the wall, then rotate it away from
+ * the wall.
+ */
+void EnBb_CheckForWall(EnBb* this) {
+    s16 yawDiff;
+
+    if (this->actor.bgCheckFlags & 8) {
+        yawDiff = this->actor.shape.rot.y - this->actor.wallYaw;
+        if (ABS_ALT(yawDiff) >= 0x4001) {
```
Comment (hensldm) [inline src/overlays/actors/ovl_En_Bb/z_en_bb.c:176]:
```suggestion
        if (ABS_ALT(yawDiff) > 0x4000) {
```
The rounder number is probably preferred.

File: src/overlays/actors/ovl_En_Bb/z_en_bb.c [hunk 2]
```diff
@@ -36,118 +47,667 @@ const ActorInit En_Bb_InitVars = {
...
+
+        if (this->actor.velocity.y < -14.0f) {
+            this->actor.velocity.y *= -0.7f;
+        } else {
+            this->actor.velocity.y = 10.0f;
+        }
+
+        this->actor.bgCheckFlags &= ~1;
+        Actor_SpawnFloorDustRing(globalCtx, &this->actor, &this->actor.world.pos, 7.0f, 2, 2.0f, 0, 0, 0);
+        Math_ScaledStepToS(&this->actor.shape.rot.y, (s16)(this->actor.yawTowardsPlayer + 0x8000), 0xBB8);
```
Comment (hensldm) [inline src/overlays/actors/ovl_En_Bb/z_en_bb.c:345]:
```suggestion
        Math_ScaledStepToS(&this->actor.shape.rot.y, BINANG_ADD(this->actor.yawTowardsPlayer, 0x8000)), 0xBB8);
```

File: src/overlays/actors/ovl_En_Bb/z_en_bb.c [hunk 3]
```diff
@@ -36,118 +47,667 @@ const ActorInit En_Bb_InitVars = {
...
+}
+
+void EnBb_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnBb* this = THIS;
+
+    EnBb_UpdateDamage(this, globalCtx);
+    this->actionFunc(this, globalCtx);
+    if ((this->actionFunc != EnBb_Dead) && (this->actionFunc != EnBb_WaitForRevive)) {
+        Actor_MoveWithGravity(&this->actor);
+        Actor_UpdateBgCheckInfo(globalCtx, &this->actor, 30.0f, 25.0f, 40.0f, 7U);
```
Comment (hensldm) [inline src/overlays/actors/ovl_En_Bb/z_en_bb.c:589]:
```suggestion
        Actor_UpdateBgCheckInfo(globalCtx, &this->actor, 30.0f, 25.0f, 40.0f, 7);
```

File: src/overlays/actors/ovl_En_Bb/z_en_bb.c [hunk 4]
```diff
@@ -36,118 +47,667 @@ const ActorInit En_Bb_InitVars = {
...
+        *dList = NULL;
+    }
+
+    return false;
+}
+
+void EnBb_PostLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3s* rot, Actor* thisx) {
+    s32 pad;
+    EnBb* this = THIS;
+    MtxF* temp_v0_4;
```
Comment (hensldm) [inline src/overlays/actors/ovl_En_Bb/z_en_bb.c:638]:
```suggestion
    MtxF* mtx;
```
Should probably name this temp. I'd go with `mtx`, but up to you. Down below looks like you did something like `currentMatrixState` for something similar.

**Review by hensldm — APPROVED** [review]
Just some minor things.

**Review by Kenix3 — APPROVED** [review]

**Pull request merged**